extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -1780,6 +1788,11 @@ private static QueryDefinition makeQueryDefinition(
           MultiStageQueryContext.getRowsPerPage(querySpec.getQuery().context())
       );
       queryToPlan = querySpec.getQuery();
+    } else if (querySpec.getDestination() instanceof ExportMSQDestination) {
+      shuffleSpecFactory = ShuffleSpecFactories.getGlobalSortWithTargetSize(
+          MultiStageQueryContext.getRowsPerPage(querySpec.getQuery().context())
+      );
+      queryToPlan = querySpec.getQuery();
```

Comment:
nit: I was wondering if MSQDestination can have a method called `getShuffleSpecFactory(int targetSize)`, and all the three destinations implement it. That will get rid of the ugly instanceof checks here.

In case there are no other pressing changes to be made post this review, I am fine with keeping it the same.

Reply:
Added